src/index.ts

```diff
@@ -119,6 +119,42 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 }, (a: string, b: string) => `${a} && ${b}`);
 
 /**
+ * @summary Compare order of versions in reverse
+ * @name rcompare
+ * @public
+ * @function
+ *
+ * @description The reverse of `.compare()`. Accepts string or null values and compares
+ * them, returning a number indicating sort order. Values are parsed for valid semver
+ * strings. Sorts an array of versions in descending order when passed to `Array.sort()`.
+ *
+ * @param {string|null} versionA - The first version to compare
+ * @param {string|null} versionB - The second version to compare
+ *
+ * @returns {number} Returns `0` if `versionA == versionB`,
+ * or `-1` if `versionA` is greater, or `1` if `versionB` is greater.
+ * null values are always weighted above string values, and string values are always
+ * weighted above valid semver values.
+ * If both values are invalid semver values, then the values are compared alphabetically.
+ *
+ * @example
+ * resinSemver.compare(null, 'Resin OS 2.0.0+rev4 (prod)'); //--> 1
+ *
+ * resinSemver.compare('Ubuntu dev', 'Resin OS 2.0.0+rev4 (prod)'); //--> 1
+ *
+ * resinSemver.compare('Version A', 'Version B'); //--> -1
+ *
+ * resinSemver.compare('Resin OS 1.16.0', 'Resin OS 2.0.0+rev4 (prod)'); //--> -1
+ *
+ * resinSemver.compare('Resin OS 2.0.0+rev4 (prod)', 'Resin OS 1.16.0'); //--> 1
```

Comment:
Haha, that is _awesome_. Yes, output like that would be amazing, and lets us skip this manual example writing entirely.

This is extremely cool. I can't overstate how cool this is. I'd love it for the SDK, and I'm sure we could come up with all sorts of other places in future too across the whole of Resin, we've got loads of docs like this that are messy and easily get inaccurate. Good encouragement for us to get useful tests in place too. Even outside Resin, I expect we could make an exciting visible open-source project from it that other people would want to use. I'm super into this.

Not sure if it's useful, but one thing that might help make the code for this a little simpler would be to hook it into the actual testing process, with a mocha reporter, rather than having to run TS and find the test file(s) yourself. There's already a built-in mocha reporter that outputs markdown with the results and body of all the tests run, which might help here: https://github.com/mochajs/mocha/blob/master/lib/reporters/markdown.js#L86-L92.